src/vs/workbench/contrib/debug/browser/debugAccessibleView.ts

```diff
@@ -0,0 +1,130 @@
+	getProvider(accessor: ServicesAccessor) {
+		const viewsService = accessor.get(IViewsService);
+		const debugService = accessor.get(IDebugService);
+		const accessibleViewService = accessor.get(IAccessibleViewService);
+		const replView = getReplView(viewsService);
+		if (!replView) {
+			return undefined;
+		}
+		const focusedElement = replView.getFocusedElement();
+		return new DebugAccessibleViewProvider(replView, focusedElement, debugService, accessibleViewService);
+	}
+}
+
+class DebugAccessibleViewProvider extends Disposable implements IAccessibleViewContentProvider {
+	public readonly id = AccessibleViewProviderId.DebugConsole;
+	private _content: string | undefined;
+	private readonly _onDidChangeContent: Emitter<void> = this._register(new Emitter<void>());
+	public readonly onDidChangeContent: Event<void> = this._onDidChangeContent.event;
+	private readonly _onDidResolveChildren: Emitter<void> = this._register(new Emitter<void>());
+	public readonly onDidResolveChildren: Event<void> = this._onDidResolveChildren.event;
+
+	public readonly verbositySettingKey = AccessibilityVerbositySettingId.DebugConsole;
+	public readonly options = {
+		type: AccessibleViewType.View
+	};
+
+	private _elementPositionMap: Map<string, Position> = new Map<string, Position>();
+
+	constructor(
+		private readonly _replView: Repl,
+		private readonly _focusedElement: IReplElement | undefined,
+		@IDebugService private readonly _debugService: IDebugService,
+		@IAccessibleViewService private readonly _accessibleViewService: IAccessibleViewService) {
+		super();
+
+	}
+	public provideContent(): string {
+		const viewModel = this._debugService.getViewModel();
+		const focusedDebugSession = viewModel?.focusedSession;
+		if (!focusedDebugSession) {
+			return 'No debug session is active.';
+		}
+		const elements = focusedDebugSession.getReplElements();
+		if (!elements.length) {
+			return 'No output in the debug console.';
+		}
+		if (!this._content) {
+			this._updateContent(elements);
+		}
+		// Content is loaded asynchronously, so we need to check if it's available or fallback to the elements that are already available.
+		return this._content ?? elements.map(e => e.toString(true)).join('\n');
+	}
+
+	public onClose(): void {
+		this._content = undefined;
+		this._replView.focusTree();
+		this._elementPositionMap.clear();
+	}
+
+	public onOpen(): void {
+		// Children are resolved async, so we need to update the content when they are resolved.
+		this._register(this.onDidResolveChildren(() => {
+			this._onDidChangeContent.fire();
+			queueMicrotask(() => {
+				if (this._focusedElement) {
+					const position = this._elementPositionMap.get(this._focusedElement.getId());
+					if (position) {
+						this._accessibleViewService.setPosition(position, true);
+					}
+				}
+			});
+		}));
+	}
+
+	private async _updateContent(elements: IReplElement[]) {
+		const dataSource = this._replView.getReplDataSource();
```

Comment:
This ReplDataSource is a tree-specific data provider. Rather than exposing this, I'd prefer you get the data more directly. You could do something like `debugViewModel.focusedSession.getReplElements()`, or, you can expose the current `IDebugSession` on the view.